lib/puppet/indirector/catalog/static_compiler.rb

```diff
@@ -90,7 +90,7 @@ def find_and_replace_metadata(host, resource, file)
   # @param metadata [Puppet::FileServing::Metadata] The metadata of the given fileserver based file
   def replace_metadata(host, resource, metadata)
     [:mode, :owner, :group].each do |param|
-      resource[param] ||= metadata.send(param)
+      resource[param] ||= metadata.send(param).to_s
```

Comment:
Actually this isn't quite right. They all need to be stringified but mode needs to be in octal. Otherwise we write out mode as "420" instead of "644". I can reproduce by trying to serve a directory containing a file. The parent directory has it's mode set to "420" and puppet will fail to create the child within.